### PR TITLE
3313 Add `affine` arg to the dict transform

### LINKED ISSUE #3313: The Affine transform currently does not have an affine matrix parameter
**Is your feature request related to a problem? Please describe.**
The `Affine` transform forwards its parameters to the `AffineGrid` class.
But it currently does not provide the "affine" parameter which the `AffineGrid` class also has.

This allows to directly define a affine transformation matrix instead of individual parameters for transformations like rotation.
If computations are directly done with affine transformation matrices, this parameter is really helpful (necessary).

**Describe the solution you'd like**
The only change necessary, is to add the `affine` parameter and to forward it to the `AffineGrid` class.

```python
class Affine(Transform):
    ...
    def __init__(
        ...
        affine: Optional[NdarrayOrTensor] = None,
        ...
    ) -> None:
        ...
        self.affine_grid = AffineGrid(
            ...
            affine=affine,
            ...
        )
```

**Describe alternatives you've considered**
We patched the affine transform locally.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #3313 .

### Description
This PR is follow up of #3318 , added the `affine` arg to the dict level transform.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/spatial/dictionary.py [hunk 1]
```diff
@@ -662,6 +666,7 @@ def __init__(
             shear_params=shear_params,
             translate_params=translate_params,
             scale_params=scale_params,
+            affine=affine,
```
Comment (yiheng-wang-nv) [inline monai/transforms/spatial/dictionary.py:669]: Thanks @Nic-Ma . Could you also add unittest for it, and also for `class Affine`?

Comment (yiheng-wang-nv) [reply monai/transforms/spatial/dictionary.py:669]: It seems `class AffineGrid` also misses the corresponding unit test for `affine` parameter, could you please add it as well?

Comment (Nic-Ma) [reply monai/transforms/spatial/dictionary.py:669]: Sure, I will add it soon.

Thanks for your finding.

**Review by yiheng-wang-nv — APPROVED** [review]

File: monai/transforms/spatial/dictionary.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/transforms/spatial/dictionary.py:669]: Added unit tests in the latest commit.

Thanks.

**Pull request merged**